tests/views/test_collection_exercise.py

```diff
@@ -198,503 +142,260 @@ class File:
-            self.eq_collection_instrument,
-            self.eq_ci_selectors,
-        )
-        mock_request.get(url_get_by_survey_with_ref_start_date, json=collection_exercise_eq_ref_start_date)
-        mock_request.get(url_get_by_survey_with_ref_end_date, json=collection_exercise_eq_ref_end_date)
 
         response = self.client.get(f"/surveys/{short_name}/{period}", follow_redirects=True)
+
         self.assertIn("Set as ready for live".encode(), response.data)
 
-    @requests_mock.mock()
+    @mock_decorator(CE_READY, EQ_SURVEY, EQ_CI, SAMPLE_SUMMARY, CE_EVENTS)
     @patch("response_operations_ui.controllers.collection_instrument_controllers.get_response_json_from_service")
-    def test_collection_exercise_view_cir_count_invalid(self, mock_request, registry_instrument_count):
+    def test_cir_count_invalid(self, registry_instrument_count):
         self.app.config["CIR_ENABLED"] = True
         registry_instrument_count.return_value = {"registry_instrument_count": 0}
-        self.load_eq_survey(
-            mock_request,
-            self.eq_survey_dates,
-            self.collection_exercises,
-            collection_exercise_eq_both_ref_date["collection_exercise"],
-            self.collection_exercise_ref_both_date,
-            sample_summary_id,
-            self.sample_summary,
-            self.eq_collection_instrument,
-            self.eq_ci_selectors,
-        )
-        mock_request.get(url_get_by_survey_with_ref_start_date, json=collection_exercise_eq_ref_start_date)
-        mock_request.get(url_get_by_survey_with_ref_end_date, json=collection_exercise_eq_ref_end_date)
 
         response = self.client.get(f"/surveys/{short_name}/{period}", follow_redirects=True)
-        self.assertNotIn("Set as ready for live".encode(), response.data)
 
-    @requests_mock.mock()
-    def test_collection_exercise_view_with_pre_population(self, mock_request):
-        self.load_eq_survey(
-            mock_request,
-            self.eq_survey_dates,
-            self.collection_exercises,
-            ce_with_pre_population["collection_exercise"],
-            self.collection_exercise_ref_both_date,
-            sample_summary_id,
-            self.sample_summary,
-            self.eq_collection_instrument,
-            self.eq_ci_selectors,
-        )
+        self.assertNotIn("Set as ready for live".encode(), response.data)
 
+    @mock_decorator(CE_PREPROP, EQ_SURVEY, EQ_CI, SAMPLE_SUMMARY, CE_EVENTS)
+    def test_collection_exercise_view_with_pre_population(self):
         response = self.client.get(f"/surveys/{short_name}/{period}", follow_redirects=True)
 
         self.assertEqual(response.status_code, 200)
         self.assertIn("Pre-Populated data is available for this sample".encode(), response.data)
 
-    @requests_mock.mock()
-    @patch("response_operations_ui.controllers.sample_controllers.sample_summary_state_check_required")
-    def test_collection_exercise_update_ce_details(self, mock_request, mock_details):
```

Comment:
There is a lot of repetition in these test, if I see the same one, covering the same lines I have removed. The key is to look at line coverage to make sure I have missed anything